### DISCUSSION TITLE
Using other solver

### DISCUSSION THREAD

**Discussion by zhejiangzhoulei** [ds4dm/ml4co-competition#23] [Q&A] [answered]
Hi,
We are also wondering whether it is allowed to use other lp/mip solver except pyscipopt in our agent.

Best

**Comment by gasse** [comment, marked as answer]
Hi @zhejiangzhoulei ,

Yes that is allowed, with one restriction: commercial solvers are prohibited. See the [rules](https://www.ecole.ai/2021/ml4co-competition/#rules).

> we encourage original contributions by specifically making the use of highly-engineered commercial solvers prohibited.

Best,
Maxime

**Reply by zhejiangzhoulei** [reply]
hi @gasse, thank you for your response :)
